<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>Yen-Lin Chen</title> <meta name="author" content="Yen-Lin Chen"> <meta name="description" content="A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design. "> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css"> <link rel="canonical" href="https://jipq6175.github.io/"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js"></script> <script src="/assets/js/dark_mode.js"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item active"> <a class="nav-link" href="/">About<span class="sr-only">(current)</span></a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">Blogs</a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">Publications</a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">CV</a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fas fa-moon"></i> <i class="fas fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5"> <div class="post"> <header class="post-header"> <h1 class="post-title"> <span class="font-weight-bold">Yen-Lin</span> Chen </h1> <p class="desc">ML Scientist | Physicist | Cornell AEP PhD</p> </header> <article> <div class="profile float-right"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/prof_pic-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/prof_pic-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/prof_pic-1400.webp"></source> <img src="/assets/img/prof_pic.jpg" class="img-fluid z-depth-1 rounded" width="auto" height="auto" alt="prof_pic.jpg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div class="clearfix"> <p>I am a machine learning scientist at <a href="https://seismictx.com/" rel="external nofollow noopener" target="_blank">Seismic Therapeutic</a>, a machine learning immunology company, building models that are aware of symmetries, geometries and physics. I am currently working at the intersection of graphs, optimization, equivariant representations and 3D structure modeling plus some diffusion. I code in <a href="https://julialang.org/" rel="external nofollow noopener" target="_blank">Julia</a>, Python and C++.</p> <p>I got my PhD in <a href="https://www.aep.cornell.edu/aep" rel="external nofollow noopener" target="_blank">Applied Physics from Cornell University</a>, supervised by <a href="https://pollack.research.engineering.cornell.edu/" rel="external nofollow noopener" target="_blank">Prof. Lois Pollack</a> with a minor in Applied Statistics. I studied nucleic acid structures and dynamics with X-ray scattering and computational/ML integrations. Some of my work was published in <a href="https://doi.org/10.1073/pnas.2026656118" rel="external nofollow noopener" target="_blank">PNAS</a> and <a href="https://doi.org/10.1126/sciadv.abf6106" rel="external nofollow noopener" target="_blank">Science Advances</a>. The recording of my virtual defense can be found <a href="https://www.youtube.com/watch?v=uaiaH-bjD10" rel="external nofollow noopener" target="_blank">here</a>.</p> <p>After PhD, I joined <a href="https://www.revivemed.io/" rel="external nofollow noopener" target="_blank">ReviveMed</a>, an AI-driven drug discovery company, as a (senior) computational biologist, working on knowledge graphs, optimizations, statistical testing of graphs, graph neural networks and biomarker discovery.</p> <p>I enjoy solving puzzles; <a href="https://adventofcode.com/" rel="external nofollow noopener" target="_blank">Advent of Code</a> is one of my favorite. <a href="https://github.com/jipq6175/advent_of_code_2021" rel="external nofollow noopener" target="_blank">Here</a> are example solutions in Julia for 2021. I also enjoy mathematical derivations, e.g. LASSO oracle inequality (<a href="https://github.com/jipq6175/MATH7740/blob/master/SL-yc2253.pdf" rel="external nofollow noopener" target="_blank">writeup</a>, <a href="https://github.com/jipq6175/MATH7740/blob/master/simulation.jl" rel="external nofollow noopener" target="_blank">simulation</a>).</p> <p>I am always creating, hacking, learning, and exploring.</p> <hr> </div> <h2><a href="/publications/" style="color: inherit;">selected publications</a></h2> <div class="publications"> <ol class="bibliography"> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/iucrj-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/iucrj-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/iucrj-1400.webp"></source> <img src="/assets/img/publication_preview/iucrj.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="iucrj.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Chen:yu5020" class="col-sm-8"> <div class="title">Machine learning deciphers structural features of RNA duplexes measured with solution X-ray scattering</div> <div class="author"> Yen-Lin Chen, and Lois Pollack</div> <div class="periodical"> <em>IUCrJ</em>, Sep 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1107/S2052252520008830" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/yu5020.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> <a href="https://github.com/jipq6175/IUCrJ_MLWAXS" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1107/S2052252520008830" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1107/S2052252520008830" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Macromolecular structures can be determined from solution X-ray scattering. Small-angle X-ray scattering (SAXS) provides global structural information on length scales of 10s to 100s of Ångstroms, and many algorithms are available to convert SAXS data into low-resolution structural envelopes. Extension of measurements to wider scattering angles (WAXS or wide-angle X-ray scattering) can sharpen the resolution to below 10Å, filling in structural details that can be critical for biological function. These WAXS profiles are especially challenging to interpret because of the significant contribution of solvent in addition to solute on these smaller length scales. Based on training with molecular dynamics generated models, the application of extreme gradient boosting (XGBoost) is discussed, which is a supervised machine learning (ML) approach to interpret features in solution scattering profiles. These ML methods are applied to predict key structural parameters of double-stranded ribonucleic acid (dsRNA) duplexes. Duplex conformations vary with salt and sequence and directly impact the foldability of functional RNA molecules. The strong structural periodicities in these duplexes yield scattering profiles with rich sets of features at intermediate-to-wide scattering angles. In the ML models, these profiles are treated as 1D images or features. These ML models identify specific scattering angles, or regions of scattering angles, which correspond with and successfully predict distinct structural parameters. Thus, this work demonstrates that ML strategies can integrate theoretical molecular models with experimental solution scattering data, providing a new framework for extracting highly relevant structural information from solution experiments on biological macromolecules.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/fm.gif-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/fm.gif-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/fm.gif-1400.webp"></source> <img src="/assets/img/publication_preview/fm.gif" class="preview z-depth-1 rounded" width="auto" height="auto" alt="fm.gif" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Chen_2020_3" class="col-sm-8"> <div class="title">In Vitro Electron Density Refinement from Solution X-ray Scattering in the Wide-Angle Regime</div> <div class="author"> Yen-Lin Chen, and Lois Pollack</div> <div class="periodical"> <em>arXiv</em>, Sep 2020 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://arxiv.org/abs/2012.13370" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://github.com/jipq6175/FM" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.48550/ARXIV.2012.13370" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.48550/ARXIV.2012.13370" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>We present Frequency Marching, FM, an algorithm that refines three-dimensional electron density distributions from solution X-ray scattering data in both the small- and wide-angle regimes. This algorithm is based on a series of optimization steps, marching along the frequency (reciprocal) space and refining detailed periodic structures with the corresponding real-space resolution. Buffer subtraction and excluded volumes, key factors in extracting the signatures of the biomolecule of interest from the sample, are accounted for using implicit density models. We provide the numerical and analytical basis of the FM algorithm. We demonstrate this technique by application to structured and unstructured nucleic acid systems, where higher resolution features are carved out of low resolution reconstructions as the algorithm marches into wider angles.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/pnas-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/pnas-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/pnas-1400.webp"></source> <img src="/assets/img/publication_preview/pnas.jpeg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="pnas.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Torabi_2021" class="col-sm-8"> <div class="title">Structural analyses of an RNA stability element interacting with poly(A)</div> <div class="author"> Seyed-Fakhreddin Torabi, Yen-Lin Chen, Kaiming Zhang, and <span class="more-authors" title="click to view 8 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '8 more authors' ? 'Jimin Wang, Suzanne J. DeGregorio, Anand T. Vaidya, Zhaoming Su, Suzette A. Pabit, Wah Chiu, Lois Pollack, Joan A. Steitz' : '8 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">8 more authors</span> </div> <div class="periodical"> <em>Proceedings of the National Academy of Sciences</em>, Sep 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.pnas.org/doi/abs/10.1073/pnas.2026656118" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/polya.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1073/pnas.2026656118" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1073/pnas.2026656118" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>Cis-acting RNA elements are crucial for the regulation of polyadenylated RNA stability. The element for nuclear expression (ENE) contains a U-rich internal loop flanked by short helices. An ENE stabilizes RNA by sequestering the poly(A) tail via formation of a triplex structure that inhibits a rapid deadenylation-dependent decay pathway. Structure-based bioinformatic studies identified numerous ENE-like elements in evolutionarily diverse genomes, including a subclass containing two ENE motifs separated by a short double-helical region (double ENEs [dENEs]). Here, the structure of a dENE derived from a rice transposable element (TWIFB1) before and after poly(A) binding (∼24 kDa and ∼33 kDa, respectively) is investigated. We combine biochemical structure probing, small angle X-ray scattering (SAXS), and cryo‐electron microscopy (cryo-EM) to investigate the dENE structure and its local and global structural changes upon poly(A) binding. Our data reveal 1) the directionality of poly(A) binding to the dENE, and 2) that the dENE-poly(A) interaction involves a motif that protects the 3ʹ-most seven adenylates of the poly(A). Furthermore, we demonstrate that the dENE does not undergo a dramatic global conformational change upon poly(A) binding. These findings are consistent with the recently solved crystal structure of a dENE+poly(A) complex [S.-F. Torabi et al., Science 371, eabe6523 (2021)]. Identification of additional modes of poly(A)–RNA interaction opens new venues for better understanding of poly(A) tail biology.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/sciadv.gif-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/sciadv.gif-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/sciadv.gif-1400.webp"></source> <img src="/assets/img/publication_preview/sciadv.gif" class="preview z-depth-1 rounded" width="auto" height="auto" alt="sciadv.gif" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="He2021" class="col-sm-8"> <div class="title">The structural plasticity of nucleic acid duplexes revealed by WAXS and MD</div> <div class="author"> Weiwei He, Yen-Lin Chen, Lois Pollack, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Serdal Kirmizialtin' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Science Advances</em>, Sep 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.science.org/doi/abs/10.1126/sciadv.abf6106" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/waxsmd.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="10.1126/sciadv.abf6106" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="10.1126/sciadv.abf6106" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>The structural diversity of nucleic acid duplexes reflects the critical roles of sequence, ions, and waters. Double-stranded DNA (dsDNA) and RNA (dsRNA) helices display an unusual structural diversity. Some structural variations are linked to sequence and may serve as signaling units for protein-binding partners. Therefore, elucidating the mechanisms and factors that modulate these variations is of fundamental importance. While the structural diversity of dsDNA has been extensively studied, similar studies have not been performed for dsRNA. Because of the increasing awareness of RNA’s diverse biological roles, such studies are timely and increasingly important. We integrate solution x-ray scattering at wide angles (WAXS) with all-atom molecular dynamics simulations to explore the conformational ensemble of duplex topologies for different sequences and salt conditions. These tightly coordinated studies identify robust correlations between features in the WAXS profiles and duplex geometry and enable atomic-level insights into the structural diversity of DNA and RNA duplexes. Notably, dsRNA displays a marked sensitivity to the valence and identity of its associated cations.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/crps-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/crps-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/crps-1400.webp"></source> <img src="/assets/img/publication_preview/crps.jpeg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="crps.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="Chen2022" class="col-sm-8"> <div class="title">Insights into the structural stability of major groove RNA triplexes by WAXS-guided MD simulations</div> <div class="author"> Yen-Lin Chen, Weiwei He, Serdal Kirmizialtin, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Lois Pollack' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Cell Reports Physical Science</em>, Sep 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S2666386422002570" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/rnatpx.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-doi="https://doi.org/10.1016/j.xcrp.2022.100971" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span> <span class="__dimensions_badge_embed__" data-doi="https://doi.org/10.1016/j.xcrp.2022.100971" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 6px;"></span> </div> <div class="abstract hidden"> <p>RNA triple helices are commonly observed tertiary motifs that are associated with critical biological functions, including signal transduction. Because the recognition of their biological importance is relatively recent, their full range of structural properties has not yet been elucidated. The integration of solution wide-angle X-ray scattering (WAXS) with molecular dynamics (MD) simulations, described here, provides a new way to capture the structures of major-groove RNA triplexes that evade crystallographic characterization. This method yields excellent agreement between measured and computed WAXS profiles and allows for an atomically detailed visualization of these motifs. Using correlation maps, the relationship between well-defined features in the scattering profiles and real space characteristics of RNA molecules is defined, including the subtle conformational variations in the double-stranded RNA upon the incorporation of a third strand by base triples. This readily applicable approach has the potential to provide insight into interactions that stabilize RNA tertiary structure that enables function.</p> </div> </div> </div> </li> </ol> </div> <div class="social"> <div class="contact-icons"> <a href="mailto:%79%63%32%32%35%33@%63%6F%72%6E%65%6C%6C.%65%64%75" title="email"><i class="fas fa-envelope"></i></a> <a href="https://scholar.google.com/citations?user=tNrpPbcAAAAJ" title="Google Scholar" rel="external nofollow noopener" target="_blank"><i class="ai ai-google-scholar"></i></a> <a href="https://github.com/jipq6175" title="GitHub" rel="external nofollow noopener" target="_blank"><i class="fab fa-github"></i></a> <a href="https://www.linkedin.com/in/ychen2253" title="LinkedIn" rel="external nofollow noopener" target="_blank"><i class="fab fa-linkedin"></i></a> </div> <div class="contact-note"> </div> </div> </article> </div> </div> <footer class="fixed-bottom"> <div class="container mt-0"> © Copyright 2025 Yen-Lin Chen. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a>. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. Last updated: June 27, 2025. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js"></script> <script defer src="/assets/js/common.js"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>